### PR TITLE
pre 1.1 fixes

### PR DESCRIPTION
**Pull request by franz**
few fixes for LLVM 6.0 + random stuff
... with recent LLVM 6 pull, all tests now pass.

### CODE REVIEW

File: include/pocl_cache.h [hunk 1]
```diff
@@ -48,16 +48,6 @@ pocl_cache_create_program_cachedir(cl_program program, unsigned device_i,
```
Comment (pjaaskel) [inline include/pocl_cache.h]: Based on your commit description: What happens if thread A manages to build the kernel and start running it while thread B is simultaneously building the same kernel and then renames it, overwriting the .so loaded and being executed by A?

**Comment by pjaaskel** [comment]
474 - conformance_printf (Failed) with LLVM 5.0 after pulling this in.

File: include/pocl_cache.h (hunk 1 above)
Comment (franz) [reply include/pocl_cache.h]: Overwriting the .so should be safe, but i just realized we might be loading the same (or two different b/c of rewrite) kernel.so files with dlopen(). I'll add a commit to fix up dlcache handling a bit.

**Comment by franz** [comment]
OK, rebased. Fixed printf by dropping the offending commit, added a comit to fix dlhandle_cache.

**Pull request merged**